samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -39,7 +40,8 @@ public AbstractApplicationRunner(Config config) {
   @Override
   public StreamSpec streamFromConfig(String streamId) {
     StreamConfig streamConfig = new StreamConfig(config);
-    String physicalName = streamConfig.getPhysicalName(streamId, streamId);
+    String defaultPhysicalName = String.format("%s-%s-%s", config.get(JobConfig.JOB_NAME()), config.get(JobConfig.JOB_ID(), "1"), streamId);
+    String physicalName = streamConfig.getPhysicalName(streamId, defaultPhysicalName);
```

Comment:
Issue: This method signature feels a little weird. Why pass the default name through via the method call? Can't the default be either in the config class itself, or used only by the caller?